### DISCUSSION TITLE
Resetting the catalog.

### DISCUSSION THREAD

**Discussion by PhillipsOwen** [TerriaJS/terriajs#6516] [General]
I have been experimenting with [this iframe example](https://docs.terria.io/guide/deploying/controlling-in-an-iframe-or-popup/) of catalog loading with the goal of populating the catalog programmatically using a web service (json). I am noticing that reloading the data catalog object can produce duplicate members.

Is there a way to remove the data object (client side) before a reload of catalog data is initiated?

I have a proof of concept .html page that i can share (upon request).

Thanks!

**Comment by tephenavies** [comment]
Hi @PhillipsOwen. Does this problem still occur if each entry in the catalog is given an `id` property?

**Comment by PhillipsOwen** [comment]
Thanks so much for the response!

there is an id for every catalog as well as member definitions. note there can be multiple catalogs in a response from the DB, each with a set of member layers.

here is an example data file. this file represents the output of a DB stored function call that that is loaded into terria by a parent frame.

[temp.json.txt](https://github.com/TerriaJS/terriajs/files/9444500/temp.json.txt)
